### PR TITLE
feat: Pageheader  new style and  props

### PR DESCRIPTION
**Pull request by chenshuai2144**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

又双叒叕 改动了样式，减少 breakchange 

4EA5C9FB-27AB-4E46-9454-ECA080E55C7A

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Pageheader style adjustment      |
| 🇨🇳 Chinese |    PageHeader 的样式调整        |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/page-header/demo/actions.md](https://github.com/ant-design/ant-design/blob/pageheader/components/page-header/demo/actions.md)
[View rendered components/page-header/demo/breadcrumb.md](https://github.com/ant-design/ant-design/blob/pageheader/components/page-header/demo/breadcrumb.md)
[View rendered components/page-header/demo/content.md](https://github.com/ant-design/ant-design/blob/pageheader/components/page-header/demo/content.md)
[View rendered components/page-header/demo/pure.md](https://github.com/ant-design/ant-design/blob/pageheader/components/page-header/demo/pure.md)
[View rendered components/page-header/demo/responsive.md](https://github.com/ant-design/ant-design/blob/pageheader/components/page-header/demo/responsive.md)
[View rendered components/page-header/index.en-US.md](https://github.com/ant-design/ant-design/blob/pageheader/components/page-header/index.en-US.md)
[View rendered components/page-header/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/pageheader/components/page-header/index.zh-CN.md)

### CODE REVIEW

File: components/config-provider/index.en-US.md [hunk 1]
```diff
@@ -43,3 +43,4 @@ Some component use dynamic style to support wave effect. You can config `csp` pr
 | getPopupContainer | to set the container of the popup element. The default is to create a `div` element in `body`. | Function(triggerNode) | `() => document.body` | 3.11.0 |
 | locale | language package setting, you can find the packages in [antd/es/locale](http://unpkg.com/antd/es/locale/) | object | 3.21.0 |
 | prefixCls | set prefix class | string | ant | 3.12.0 |
+| pageHeaderType | Unify the type of pageHeader | string | ant | 3.12.0 |
```
Comment (afc163) [inline components/config-provider/index.en-US.md]: 3.12.0？

File: components/config-provider/index.zh-CN.md [hunk 2]
```diff
@@ -44,3 +44,4 @@ return (
 | getPopupContainer | 弹出框（Select, Tooltip, Menu 等等）渲染父节点，默认渲染到 body 上。 | Function(triggerNode) | () => document.body | 3.11.0 |
 | locale | 语言包配置，语言包可到 [antd/es/locale](http://unpkg.com/antd/es/locale/) 目录下寻找 | object | - | 3.21.0 |
 | prefixCls | 设置统一样式前缀 | string | ant | 3.12.0 |
+| pageHeaderType | 统一设置 pageHeader 的 type | string | ant | 3.12.0 |
```
Comment (zombieJ) [inline components/config-provider/index.zh-CN.md]:
```tsx
pageHeader?: { type }
```

抽到 `pageHeader` 对象里，和 4.0 命名对齐。

File: components/config-provider/index.en-US.md (hunk 1 above)
Comment (zombieJ) [reply components/config-provider/index.en-US.md]: 和 Button 对齐：`ghost`

File: components/page-header/demo/pure.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      </Descriptions>
+    </PageHeader>
+  </div>,
+  mountNode,
+);
+```
+
+<style>
+#components-page-header-demo-pure .code-box-demo .ant-page-header {
+  border: 1px solid #ebedf0;
```
Comment (afc163) [inline components/page-header/demo/pure.md]: 这个直接写 style 里去吧，可以无脑拷贝。

File: components/page-header/demo/pure.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
+---
+order: 2
+title:
+  zh-CN: 纯净模式，没有白底
+  en-US: Pure mode, no white background
+---
+
+## zh-CN
+
+在某些情况下，pageHeader 不需要自己的背景颜色。
```
Comment (afc163) [inline components/page-header/demo/pure.md]:
```diff
- pageHeader
+ PageHeader
```

File: components/page-header/__tests__/__snapshots__/demo.test.js.snap [hunk 5]
```diff
@@ -182,7 +182,7 @@ exports[`renders ./components/page-header/demo/actions.md correctly 1`] = `
   </div>
   <br />
   <div
-    class="ant-page-header"
+    class="ant-page-header ghost"
```
Comment (afc163) [inline components/page-header/__tests__/__snapshots__/demo.test.js.snap]: 应该是 `ant-page-header-ghost`

File: components/page-header/demo/pure.md [hunk 6]
```diff
@@ -1,17 +1,17 @@
...
-  zh-CN: 纯净模式，没有白底
-  en-US: Pure mode, no white background
+  zh-CN: 白底模式
+  en-US: white background mode
 ---
 
 ## zh-CN
 
-在某些情况下，pageHeader 不需要自己的背景颜色。
+在某些情况下，pageHeader 需要自己的背景颜色。
```
Comment (afc163) [inline components/page-header/demo/pure.md]:
```diff
- pageHeader
+ PageHeader
```

Comment (afc163) [reply components/page-header/demo/pure.md]: 默认 PageHeader 是透明底色的。在某些情况下，PageHeader 需要自己的背景颜色。

File: components/page-header/demo/responsive.md [hunk 7]
```diff
@@ -92,8 +92,9 @@ ReactDOM.render(
 tr:last-child td {
   padding-bottom: 0;
 }
-.ant-statistic-content-value{
+.ant-statistic-content{
```
Comment (afc163) [inline components/page-header/demo/responsive.md]: { 前少空格。

File: components/page-header/demo/pure.md [hunk 8]
```diff
@@ -55,6 +55,6 @@ ReactDOM.render(
 
 <style>
 #components-page-header-demo-pure .code-box-demo .ant-page-header {
-  border: 1px solid #ebedf0;
+  border: none;
```
Comment (afc163) [inline components/page-header/demo/pure.md]: pure.md 比较奇怪。改一下。

Comment (afc163) [reply components/page-header/demo/pure.md]: PageHeader 的 border 应该本来就是 none 吧？

Comment (chenshuai2144) [reply components/page-header/demo/pure.md]: 被第一个设置上了，这里需要重置一下

Comment (afc163) [reply components/page-header/demo/pure.md]: 各管各的，第一个设置自己的，不要重置。

**Comment by chenshuai2144** [comment]
设计师总想搞点微调，好累😫

File: components/config-provider/index.tsx [hunk 9]
```diff
@@ -63,6 +68,12 @@ class ConfigProvider extends React.Component<ConfigProviderProps> {
       config.renderEmpty = renderEmpty;
     }
 
+    // set pageHeader type
+    // 防止设置为 null 报错
```
Comment (zombieJ) [inline components/config-provider/index.tsx]: 这个判断挪到 PageHeader 里，万一外层没有 ConfigProvider。

File: components/page-header/index.tsx [hunk 10]
```diff
@@ -104,6 +106,8 @@ const PageHeader: React.SFC<PageHeaderProps> = props => (
       const className = classnames(prefixCls, customizeClassName, {
         'has-breadcrumb': breadcrumbDom,
         'has-footer': footer,
+        // 防止设置为 null 报错
+        [`${prefixCls}-ghost`]: ghost || (pageHeader && ghost === undefined && pageHeader.ghost),
```
Comment (zombieJ) [inline components/page-header/index.tsx]: 'ghost' in props ? ghost : pageHeader.ghost;

File: components/page-header/index.tsx [hunk 11]
```diff
@@ -89,7 +90,7 @@ const renderChildren = (prefixCls: string, children: React.ReactNode) => {
 
 const PageHeader: React.SFC<PageHeaderProps> = props => (
   <ConfigConsumer>
-    {({ getPrefixCls }: ConfigConsumerProps) => {
+    {({ getPrefixCls, pageHeader = { ghost: true } }: ConfigConsumerProps) => {
```
Comment (zombieJ) [inline components/page-header/index.tsx]: 这个可以更新掉了~

File: components/page-header/index.tsx [hunk 12]
```diff
@@ -98,12 +99,24 @@ const PageHeader: React.SFC<PageHeaderProps> = props => (
...
+      } else if (pageHeader && 'ghost' in pageHeader) {
+        ghost = pageHeader.ghost;
+      }
 
       const prefixCls = getPrefixCls('page-header', customizePrefixCls);
       const breadcrumbDom = breadcrumb && breadcrumb.routes ? renderBreadcrumb(breadcrumb) : null;
       const className = classnames(prefixCls, customizeClassName, {
         'has-breadcrumb': breadcrumbDom,
         'has-footer': footer,
+        // 防止设置为 null 报错
```
Comment (zombieJ) [inline components/page-header/index.tsx]: 去掉

File: components/page-header/index.tsx [hunk 13]
```diff
@@ -98,12 +99,23 @@ const PageHeader: React.SFC<PageHeaderProps> = props => (
         breadcrumb,
         className: customizeClassName,
       } = props;
+      let ghost: undefined | boolean = true;
+
+      // If there is ghost in props, use props
```
Comment (zombieJ) [inline components/page-header/index.tsx]: ``` Use `ghost` from `props` or from `ConfigProvider` instead. ```

**Review by zombieJ — DISMISSED** [review]

**Review by zombieJ — APPROVED** [review]

**Pull request merged**